### ISSUE TITLE
Type Checking in Parser

### ISSUE THREAD

**Issue by jrgorecki** [jrgorecki/Project#8]
Ok, so it looks like I gotta remove all the convenient type casting done in the parser. The typechecker knows that expr + expr is an IntT, expr <> expr as BoolT,  etc, correct? 

All the parser defines in terms of types is stuff like (Con 0, UnitT)

**Comment by jrgorecki** [comment]
It isn't clear that we will lose points due to inferred types in the parser, but Cesare appeared to be wholly against the notion. Prof. Tinelli said, "You should not do type checking in the parser, regardless of whether you are going to be penalized or not."

Let me know if you think we should remove typing from the parser. 
   I think no, since he was not explicitly against it. But it's understandable to not include them if you're worried.

**Comment by dougives** [comment]
Yep, I forgot to add `| (Con(v), t), env -> Con(v), t` to the typechecker. The tests seemed designed to fail, so we have to write some working ones to test. So far it seems good, I'll add that little fix.

I think he's more worried that we might try to walk through the AST to assign types during parsing or something of the sort. What we're doing seems sensible to me.

**Comment by dougives** [comment]
Actually I just realized that we're supposed to remove them one by one, in order to check that no funny business is happening.
